Budget frontend

klaidas01 opened the pull request.

klaidas01 committed:
add budget creation form with validation

klaidas01 committed:
create progress bar component

klaidas01 committed:
make progress bar component centered

klaidas01 committed:
create budget list item

MantasGra committed:
Add budgets CRUD

klaidas01 committed:
add currentAmount to budget model

klaidas01 committed:
Merge branch 'budget-backend' into budget-frontend

klaidas01 committed:
connect frontend with backend, create budget list view

klaidas01 committed:
add a message which is displayed when no budgets are found

Martynas-S reviewed (commented):
Kodel 01bff7850592eaadf85e38aa30bd7881dda07df6 daro changes `package-lock.json` failam?

klaidas01 commented:
> 
> 
> Kodel [01bff78](https://github.com/moneybunney/MoneyBunney/commit/01bff7850592eaadf85e38aa30bd7881dda07df6) daro changes `package-lock.json` failam?

Net neisivaizduoju, cia nebuvo intended

Martynas-S commented on frontend/src/Components/BudgetList/BudgetList.tsx:
```diff
@@ -0,0 +1,77 @@
+  const collapsedHeight = loading
+    ? `${spinnerSize + marginTopSize * 2}px`
+    : undefined;
+  if (budgets.length == 0) {
```
Cia turetu buti `  if (!loading && budgets.length === 0) {`

Martynas-S commented on frontend/src/Components/BudgetList/BudgetListItem.tsx:
```diff
@@ -0,0 +1,64 @@
+  const parsedStartDate = new Date(budget.startDate);
+  const parsedEndDate = new Date(budget.endDate);
+
+  const startDateString = toDisplayDate(parsedStartDate).split(" ")[0];
```
Jeigu tu pats konstruoji data su laiku ir paskui naudoji tik data, be laiko, tai kam tuomet skaiciuoti ta laika in the first place, atskirti ji per tarpa ir iskart splittint? :D

Martynas-S commented on frontend/src/Components/BudgetList/BudgetListItem.tsx:
```diff
@@ -0,0 +1,64 @@
+  const [shown, setShown] = React.useState(false);
+
+  useEffect(() => {
+    setShown(true);
```
Sitas praktiskai nieko kitaip nedaro nei `const [shown, setShown] = React.useState(true);`

Martynas-S reviewed (approved):
Viskas kaip ir gerai, tik pora change'u reiktu padaryt:
- Kai loadina budgetus, nerodo jokio spinnerio, nes `(budgets.length == 0)`
- Kazkodel budget list item'ai yra mygtukai, nors juos paspaudus nieko nedaro ir neturetu daryt :D
- Dar manau, kad butu gerai, jog kol viskas ok su budgetu jis butu zalias, artejan pasidarytu geltonas, o raudonas butu tik virsijus
- Dar butu gerai, kad parysketu budget amount tekstas kai virsijamas budgetas

Martynas-S commented on frontend/src/Components/BudgetList/BudgetListItem.tsx:
```diff
@@ -0,0 +1,64 @@
+  load
+}: IProps) => {
+  const classes = useStyles();
+  const [shown, setShown] = React.useState(false);
```
Oi sori, nepamaciau, kad sitas state is vis prasmes neturi, nieko nedaro :D

Martynas-S committed:
Fix up <BudgetListItem /> component

- Extract only the needed part of the data in the toDisplayDate()
  function
- Remove unused state `shown`
- Remove unecessary `<Collapse />`
- Make `<ListItem />` not be a button for no reason

Martynas-S committed:
Fix <BudgetListItem />

- tslint formatting
- Make spinner shown when loading

Martynas-S committed:
Make BudgetList not display budgets that are out-dated

Martynas-S committed:
Add <BudgetList /> to the Dashboard
